relay-server/src/utils/rate_limits.rs

```diff
@@ -850,4 +900,69 @@ mod tests {
         mock.assert_call(DataCategory::Attachment, None);
         mock.assert_call(DataCategory::Session, None);
     }
+
+    #[test]
+    fn test_enforce_transaction_no_metrics_extracted() {
+        let mut envelope = envelope![Transaction];
+
+        let mut mock = MockLimiter::default().deny(DataCategory::TransactionProcessed);
+        let limiter = EnvelopeLimiter::new(|s, q| mock.check(s, q));
+        let (enforcement, limits) = limiter.enforce(&mut envelope, &scoping()).unwrap();
+
+        assert!(limits.is_limited());
+        assert!(enforcement.extracted_transaction_metrics.is_active());
+        assert!(enforcement.event.is_active());
+        mock.assert_call(DataCategory::TransactionProcessed, Some(0));
+    }
+
+    #[test]
+    fn test_enforce_transaction_metrics_extracted() {
+        let mut envelope = envelope![Transaction];
+        envelope
+            .get_item_by_mut(|item| *item.ty() == ItemType::Transaction)
+            .unwrap()
+            .set_metrics_extracted(true);
+
+        let mut mock = MockLimiter::default().deny(DataCategory::TransactionProcessed);
+        let limiter = EnvelopeLimiter::new(|s, q| mock.check(s, q));
+        let (enforcement, limits) = limiter.enforce(&mut envelope, &scoping()).unwrap();
+
+        assert!(limits.is_limited());
+        assert!(enforcement.extracted_transaction_metrics.is_active());
+        assert!(enforcement.event.is_active());
+    }
+
+    #[test]
+    fn test_enforce_transaction_no_indexing_quota() {
+        let mut envelope = envelope![Transaction];
+
+        let mut mock = MockLimiter::default().deny(DataCategory::Transaction);
+        let limiter = EnvelopeLimiter::new(|s, q| mock.check(s, q));
+        let (enforcement, limits) = limiter.enforce(&mut envelope, &scoping()).unwrap();
+
+        assert!(!limits.is_limited());
+        assert!(!enforcement.extracted_transaction_metrics.is_active());
+
+        // Note we never check for the DataCategory::Transaction so we are not setting this
+        // to enforced.  After metrics extraction will check the rate limiter again and then
+        // this will actually do something.
+        assert!(!enforcement.event.is_active());
```

Comment:
I believe this is an invalid assumption. Even if metrics extraction is switched off, we will still have to check for this quota. This check always needs to run.

Reply:
Should the indexed transaction quota check run at a specific point in the pipeline? Quotas are checked twice throughout the processing of the envelope, and the indexed transaction quota is only checked the second time. I believe the behavior is correct.

Reply:
I still believe this is a correct assertion: When we have not yet extracted metrics we are in `check_envelope`.  At this point our design said to only ever check the `TransactionProcessed` quota and we never need to check `Transaction` quota.  So the fact that `Transaction` quota is out should not result in any enforcement.

Reply:
Updated the description to include the full algorithm. Viewed in isolation, the assertion here is correct, but this case should never be triggered due to the precondition: There must always be processing transaction quota that's also checked for legacy plans. In the implementation, so far we've not updated tests to reflect this. 

@flub and I have had an offline conversation about this and will circle around to this.